test/helpers/handshake.c

```diff
@@ -1609,6 +1609,18 @@ static HANDSHAKE_RESULT *do_handshake_internal(
                         ret->result = SSL_TEST_INTERNAL_ERROR;
                         goto err;
                     }
+                } else if (client_turn && server.status == PEER_ERROR) {
+                    /*
+                     * The server Failed and result client non-block.
+                     */
+                    ret->result = SSL_TEST_SERVER_FAIL;
+                    goto err;
+                } else if (!client_turn && client.status == PEER_ERROR) {
+                    /*
+                     * The client Failed and result server non-block.
+                     */
+                    ret->result = SSL_TEST_CLIENT_FAIL;
+                    goto err;
```

Comment:
Why is this change necessary? It suggests a change of behaviour which is undesirable.

Reply:
It  behavior should be compatible and still kept with the update expected error, **The test case:** this side return failed and did not send message . and the  peer detect the non-block and then check with the prior status, this case, the root error should from this side.
The prior error code: this side:ERROR, the peer:ERROR
The current error code: this side:ERROR, the peer:RETRY

Reply:
ossl_statem_in_error could not handle the error from application protocol,this kept for compatibility.And this will remove once all other case update the statem error.